api-graphql/graph/model/exam.go

```diff
@@ -0,0 +1,53 @@
+package model
+
+type Exam interface {
```

Comment:
I agree with the inheritance relationship here, but maybe you should imploy this with a method called maybe GetYields() since that accurately unifies all of its inheriting subclasses.